generator/integration_tests/golden/v1/golden_thing_admin_rest_connection.h

```diff
@@ -0,0 +1,64 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Generated by the Codegen C++ plugin.
+// If you make any local changes, they will be lost.
+// source: generator/integration_tests/test.proto
+
+#ifndef GOOGLE_CLOUD_CPP_GENERATOR_INTEGRATION_TESTS_GOLDEN_V1_GOLDEN_THING_ADMIN_REST_CONNECTION_H
+#define GOOGLE_CLOUD_CPP_GENERATOR_INTEGRATION_TESTS_GOLDEN_V1_GOLDEN_THING_ADMIN_REST_CONNECTION_H
+
+#include "generator/integration_tests/golden/v1/golden_thing_admin_connection.h"
+#include "google/cloud/options.h"
+#include "google/cloud/version.h"
+#include <memory>
+
+namespace google {
+namespace cloud {
+namespace golden_v1 {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+/**
+ * A factory function to construct an object of type `GoldenThingAdminConnection`
+ * that uses REST over HTTP as transport in lieu of gRPC. REST transport should
+ * only be used for services that do not support gRpc or if the existing network
+ * configuration precludes using gRPC.
+ *
+ * The returned connection object should not be used directly; instead it
+ * should be passed as an argument to the constructor of GoldenThingAdminClient.
+ *
+ * The optional @p options argument may be used to configure aspects of the
+ * returned `GoldenThingAdminConnection`. Expected options are any of the types in
+ * the following option lists:
+ *
+ * - `google::cloud::CommonOptionList`
+ * - `google::cloud::RestOptionList`
+ * - `google::cloud::UnifiedCredentialsOptionList`
+ * - `google::cloud::golden_v1::GoldenThingAdminPolicyOptionList`
+ *
+ * @note Unexpected options will be ignored. To log unexpected options instead,
+ *     set `GOOGLE_CLOUD_CPP_ENABLE_CLOG=yes` in the environment.
+ *
+ * @param options (optional) Configure the `GoldenThingAdminConnection` created by
+ * this function.
+ */
+std::shared_ptr<GoldenThingAdminConnection> MakeGoldenThingAdminConnectionRest(
+    Options options = {});
```

Comment:
Can we have a unit test for this thing?

I know these are kinda weird to test... but can we do something like set a fake local endpoint, and make one client call we know will fail? or even just call `conn->options()` on the returned connection?